azbus/receiver.go

```diff
@@ -199,10 +215,38 @@ func (r *Receiver) elapsed(ctx context.Context, count int, total int, maxDuratio
 	return err
 }
 
-// RenewMessageLock renews the given messages peek lock, so it doesn't lose the lock and get re-added to the message queue.
+// elapsed emits 2 log messages detailing how long processing took. Additionally executes suitable disposition of message.
+func (r *Receiver) elapsedParallel(ctx context.Context, count int, maxDuration time.Duration, msg *ReceivedMessage, handler PHandler) {
+	now := time.Now()
+
+	log := r.log.FromContext(ctx)
+	defer log.Close()
+
+	log.Debugf("Processing message %d", count)
+	disp, ctx, err := r.handleParallelReceivedMessageWithTracingContext(ctx, msg, handler)
+	_ = r.Dispose(ctx, disp, err, msg)
+
+	duration := time.Since(now)
+	log.Debugf("Processing message %d took %s", count, duration)
+
+	// This is safe because maxDuration is only defined if RenewMessageLock is false.
+	if !r.Cfg.RenewMessageLock && duration >= maxDuration {
+		log.Infof("WARNING: processing msg %d duration %v took more than %v seconds", count, duration, maxDuration)
+		log.Infof("WARNING: please either enable SERVICEBUS_RENEW_LOCK or reduce SERVICEBUS_INCOMING_MESSAGES")
+		log.Infof("WARNING: both can be found in the helm chart for each service.")
+	}
+	if err != nil {
+		if errors.Is(err, ErrPeekLockTimeout) {
+			log.Infof("WARNING: processing msg %d duration %s returned error: %v", count, duration, err)
+			log.Infof("WARNING: please enable SERVICEBUS_RENEW_LOCK which can be found in the helm chart")
+		}
+	}
```

Comment:
no need for this outer `if err != nil` if we only do things inside the inner errors.IS if.

```suggestion
	if errors.Is(err, ErrPeekLockTimeout) {
		log.Infof("WARNING: processing msg %d duration %s returned error: %v", count, duration, err)
		log.Infof("WARNING: please enable SERVICEBUS_RENEW_LOCK which can be found in the helm chart")
	}
```

Reply:
Also we need to handle other error conditions if the error is something else.